docs/design/disaster-recovery-optimization.md

```diff
@@ -0,0 +1,52 @@
+# FEATURE: Regional Disaster Recovery Optimization
+
+## Summary
+
+Implement use of `bluestore-rdr` as the OSD backend store for customers using ODF with Regional Disaster Recovery (RDR).
+
+## Motivation
+
+- Ceph uses BlueStore as a special-purpose storage back end designed specifically for managing data on disk for Ceph OSD workloads. Ceph team is working on improving this backend storage that will help in significantly improving the performance, especially in disaster recovery scenarios. This new OSD backend is known as bluestore-rdr. In the future, other backends will also need to be supported, such as seastore.
+
+## Proposal
+
+- Enable the optimized OSD backend store for RDR clusters.
+- Enable this optimization for both new (Greenfield) and existing (brownfield) customers.
+
+### Greenfield
+- If `ocs.openshift.io/optimizeDisasterRecovery: true` annotation is available in the `StorageCluster` then configure OSD store type in the `cephCluster` as:
+
+```YAML
+spec:
+  Storage:
+    store:
+        type: bluestore-rdr
```

Comment:
> If the annotation is set, then spec.mirroring.enabled will also be true, right?

I don't think that would be the case. So user creates two ODF cluster, connects them via ACM and then enables mirroring using a DR policy in MultiClusterOrchestrator.  Its at this point when `spec.mirroring.enabled` is set to `true`. 

`spec.mirroring.enabled` won't be `true` for a new cluster ODF cluster.  So new clusters need to rely on the `annotation` only to decide if they want to use the OSD store optimizations. 

If `spec.mirroring.enabled` is `true`, then it will be a brownfield case. That cluster was already created. Now we can check from the status if optimizations are applied already or not. If not applied, then we can pass `yes-really-update-store` option to the ceph cluster.

Reply:
@umangachapagain Thoughts on this?

Reply:
> Do we even need to distinguish between green field and brown field? Seems like the only real difference is if we decide to implement green field first, and then brown field later. Really, the brown field implementation will handle green field as well.
>
Totally agree. Ideally, this should all be hidden behind ocs-operator and handled with a brownfield implementation. However, there were some concerns raised with testing and supporting brownfield in the short term.

So, we're going ahead with greenfield support only. In the long term, we'll handle it as brownfield.

> If the annotation is set, then spec.mirroring.enabled will also be true, right?
>
I agree with @sp98 here. If the annotation is set, it doesn't imply that mirroring is enabled. It only indicates that mirroring might be enabled.